flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/consumer/SingleInputGateTest.java

```diff
@@ -542,6 +543,29 @@ public void testUpdateUnknownInputChannel() throws Exception {
 		}
 	}
 
+	@Test
+	public void checkInputGateRemoveInNetworkEnvironment() throws IOException {
+		NetworkEnvironment network = createNetworkEnvironment();
+		InputChannelDeploymentDescriptor[] channelDescs = new InputChannelDeploymentDescriptor[]{
+			// Local
+			new InputChannelDeploymentDescriptor(
+				new ResultPartitionID(),
+				ResultPartitionLocation.createLocal())};
+		IntermediateDataSetID id = new IntermediateDataSetID();
+		InputGateDeploymentDescriptor igdd = new InputGateDeploymentDescriptor(id,
+			ResultPartitionType.PIPELINED, 0, channelDescs);
+		SingleInputGate[] inputGates = network.createInputGates("",
```

Comment:
For the array case, it seems better to make size more than 1, such as 2? I am not very sure.